src/cargo/core/resolver/resolve.rs

```diff
@@ -203,8 +203,19 @@ impl<'a> Iterator for Deps<'a> {
             .and_then(|e| e.next())
             .map(|id| self.resolve.replacement(id).unwrap_or(id))
     }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        match self.edges {
+            // Note: Edges is actually a std::collections::hash_set::Iter, which
+            // is an ExactSizeIterator.
+            Some(ref iter) => iter.size_hint(),
```

Comment:
Instead of a comment, let's perhaps write

```
let len = self.edges.as_ref().map(ExactSizeIterator::len).unwrap_or(0)
```

so that, if edges ceases to be `ExactSize` for whatever reason, we notice this at compile time?